packages/@uppy/react/src/useUppyState.ts

```diff
@@ -0,0 +1,23 @@
+import type { Uppy, State } from '@uppy/core'
+import type { Body, Meta } from '@uppy/utils/lib/UppyFile'
+import { useMemo, useCallback } from 'react'
+import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/with-selector'
+
+export default function useUppyState<
+  M extends Meta = Meta,
+  B extends Body = Body,
+  T = any,
```

Comment:
can `any` be avoided? maybe `unknown` is possible?

Reply:
For what it's worth, this is an inferred generic. It will always be typed to something other than `any`.

Reply:
You should be able to get away with not providing the default in that case.

Reply:
```suggestion
  T,
```

Reply:
```
Error: packages/@uppy/react/src/useUppyState.ts(9,3): error TS2706: Required type parameters may not follow optional type parameters.
```

Reply:
🤦